osu.Game/Database/DatabaseWorkingBeatmap.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.IO;
+using osu.Framework.Audio.Track;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Formats;
+using osu.Game.Beatmaps.IO;
+
+namespace osu.Game.Database
+{
+    public partial class BeatmapDatabase
+    {
+        private class DatabaseWorkingBeatmap : WorkingBeatmap
+        {

```

Comment:
Discussion point, made this a partial class just to move the nested private class to its own file